src/data-pipeline/etl-common/src/main/java/software/aws/solution/clickstream/common/sensors/SensorsEventParser.java

```diff
@@ -0,0 +1,346 @@
+            deviceUaMap.put(UA_STRING, sensorsEvent.getProperties().getUserAgent());
+            clickstreamEvent.setDeviceUa(deviceUaMap);
+        }
+
+        if (null != sensorsEvent.getProperties()) {
+            clickstreamEvent.setDeviceScreenWidth(sensorsEvent.getProperties().getScreenWidth());
+            clickstreamEvent.setDeviceScreenHeight(sensorsEvent.getProperties().getScreenHeight());
+        }
+    }
+
+    private void setGeoInfo(final SensorsEvent sensorsEvent, final ClickstreamEvent clickstreamEvent) {
+        if (sensorsEvent.getProperties() != null) {
+            clickstreamEvent.setGeoCountry(sensorsEvent.getProperties().getCity());
+        }
+    }
+
+    private Map<String, ClickstreamEventPropValue> getEventCustomParameters(final SensorsEvent sensorsEvent, final ClickstreamEvent clickstreamEvent) throws JsonProcessingException {
+        Map<String, ClickstreamEventPropValue> customParameters = new HashMap<>();
+        if (sensorsEvent.getUnknownProperties() != null) {
+            customParameters = convertStringObjectMapToStringEventPropMap(sensorsEvent.getUnknownProperties());
+            clickstreamEvent.setCustomParameters(customParameters);
+        }
+        String screenResolution = sensorsEvent.getProperties().getScreenWidth() + "x" + sensorsEvent.getProperties().getScreenHeight();
+        customParameters.put("client_id", new ClickstreamEventPropValue(sensorsEvent.getDistinctId(), ValueType.STRING));
+        customParameters.put("ip_override", new ClickstreamEventPropValue(sensorsEvent.getProperties().getIp(), ValueType.STRING));
+        customParameters.put("screen_resolution", new ClickstreamEventPropValue(screenResolution, ValueType.STRING));
+        customParameters.put("user_agent", new ClickstreamEventPropValue(sensorsEvent.getProperties().getUserAgent(), ValueType.STRING));
+
+        if (sensorsEvent.getProperties().getUserAgent() != null) {
+            customParameters.put("x-ga-ur",
+                    new ClickstreamEventPropValue(sensorsEvent.getProperties().getUserAgent(), ValueType.STRING));
+        }
+        if (sensorsEvent.getProperties().getProvince() != null && sensorsEvent.getProperties().getCity() != null) {
+            ObjectMapper objectMapper = new ObjectMapper();
+            ObjectNode eventObject = objectMapper.createObjectNode();
+            ObjectNode locationObject = objectMapper.createObjectNode();
+
+            locationObject.put("province", sensorsEvent.getProperties().getProvince());
+            locationObject.put("city", sensorsEvent.getProperties().getCity());
+
+            eventObject.set("event_location", locationObject);
+            customParameters.put("event_location",
+                    new ClickstreamEventPropValue(objectToJsonString(eventObject), ValueType.OBJECT));
+        }
+        if (sensorsEvent.getProperties() != null) {
+            customParameters.put("x-ga-system_properties",
```

Comment:
change `x-ga-system_properties` to `properties`.  customParameters only store properties that cannot be mapped to clickstream properties

Reply:
also merge `SensorsEvent.getUnknownProperties()` to  `customParameters`

Reply:
Any properties in SensorsEvent that cannot be mapped to ClickstreamEvent properties should be set to customParameters too.